src/irmin/metrics.mli

```diff
@@ -0,0 +1,83 @@
+(*
+ * Copyright (c) 2022 Etienne Marais <etienne@maiste.fr>
+ *
+ * Permission to use, copy, modify, and distribute this software for any
+ * purpose with or without fee is hereby granted, provided that the above
+ * copyright notice and this permission notice appear in all copies.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+ * WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+ * MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+ * ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+ * WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+ * ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+ * OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+ *)
+
+(** [Metrics] defines primitives to handle metrics inside of Irmin. Its purpose
+    is to decouple the metrics type definition from the data manipulation.
+
+    It has a 'a {!repr} type to describe the type of the metric manipulated.
+    This {!repr} is used to define a {!metric}. It is in charge of how it is
+    gathered and how it should be stored (if needed).
+
+    A {!metric} can be modified in different ways, depending on the {!modifier}. *)
+
+type origin = ..
+(** An extensible type to get the location of the definition. *)
+
+type 'a repr
+(** {!ty} is the {!Repr.ty} description of the type used by metrics. *)
+
+val repr : string -> 'a Repr.ty -> 'a repr
+(** [repr name t] generates a metric representation, named according to [name]. *)
```

Comment:
From the `Metrics` user point of view, is the only purpose of a `_ repr` to be passed to `create`? If true then why not inline the `string` and `a Repr.t` parameters in the `create` function and get rid of the `_ repr` type?